frontend/waitingPage.js

```diff
@@ -6,21 +6,47 @@ class Waiting extends Component {
   constructor(props) {
     super(props);
     this.state = {
-      processing: false,
+      processing: true,
+      processingAfter10: false,
       errored: false,
-      finished: true
+      finished: false,
     }
   }
 
+  componentDidMount() {
+    setTimeout(() => {
+      this.setState({ processingAfter10: true });
+    }, 10000);
+  }
+
   renderWaitingText = (props) => {
     const { navigation } = this.props;
-    if (props.processing && !props.errored) {
+    if (props.processing && !props.errored && !props.processingAfter10) {
       return (
         <View style={styles.card}>
-          <Text style={{ fontSize: 20 }}>The video is being processed.</Text>
+          <Text style={{ fontSize: 20 }}>
+            We are processing your video.
+          </Text>
           <Spinner />
         </View>
       )
+    } else if (props.processing && props.processingAfter10) {
+      return (
+        <View style={styles.card}>
+          <Text style={{ fontSize: 20 }}>
+            Sorry, we cannot generate a result right now. Please wait as we continue processing you video.
+          </Text>
+          <Spinner />
+          <Button>
+            <Text
+              onPress={function () {
+                navigation.navigate('recordingModule');
```

Comment:
navigate brings you to current state (upload option), navigation.push('recordingModule') takes you to new recording screen